packages/studio-plugin/src/ParsingOrchestrator.ts

```diff
@@ -208,15 +221,30 @@ export default class ParsingOrchestrator {
     return this.filepathToFileMetadata;
   }
 
-  private initLayoutNames(): string[] {
+  private initLayoutNameToLayoutFile(): Record<string, LayoutFile> {
     if (!fs.existsSync(this.paths.layouts)) {
-      return [];
+      return {};
     }
     const files = fs.readdirSync(this.paths.layouts, "utf-8");
-    return files.map((filename) => {
+    return files.reduce((layoutMap, filename) => {
```

Comment:
`readdirSync` does add all elements to `files` including sub-directories. I was going off the assumption that sub-directories aren't allowed in `src/layouts` just like we don't allow them for `src/pages`. But, it is more graceful to ignore those instead of throwing an error, so I updated both pages and layouts to filter out non-file elements (and updated tests too)